seatunnel-core/seatunnel-flink-starter/seatunnel-flink-13-starter/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkRuntimeEnvironment.java

```diff
@@ -247,87 +247,84 @@ private void setCheckpoint() {
             log.warn(
                     "Disabled Checkpointing. In flink execution environment, checkpointing is not supported and not needed when executing jobs in BATCH mode");
         }
-        long interval = 0;
+        long interval;
         if (config.hasPath(EnvCommonOptions.CHECKPOINT_INTERVAL.key())) {
             interval = config.getLong(EnvCommonOptions.CHECKPOINT_INTERVAL.key());
         } else if (config.hasPath(ConfigKeyName.CHECKPOINT_INTERVAL)) {
             log.warn(
                     "the parameter 'execution.checkpoint.interval' will be deprecated, please use common parameter 'checkpoint.interval' to set it");
             interval = config.getLong(ConfigKeyName.CHECKPOINT_INTERVAL);
+        } else {
+            interval = EnvCommonOptions.CHECKPOINT_INTERVAL.defaultValue();
         }
 
-        if (interval > 0) {
-            CheckpointConfig checkpointConfig = environment.getCheckpointConfig();
-            environment.enableCheckpointing(interval);
-
-            if (EnvironmentUtil.hasPathAndWaring(config, ConfigKeyName.CHECKPOINT_MODE)) {
-                String mode = config.getString(ConfigKeyName.CHECKPOINT_MODE);
-                switch (mode.toLowerCase()) {
-                    case "exactly-once":
-                        checkpointConfig.setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE);
-                        break;
-                    case "at-least-once":
-                        checkpointConfig.setCheckpointingMode(CheckpointingMode.AT_LEAST_ONCE);
-                        break;
-                    default:
-                        log.warn(
-                                "set checkpoint.mode failed, unknown checkpoint.mode [{}],only support exactly-once,at-least-once",
-                                mode);
-                        break;
-                }
-            }
+        CheckpointConfig checkpointConfig = environment.getCheckpointConfig();
```

Comment:
How about doing some refactoring? Currently there are two copies of the same processing logic.

Reply:
@Hisoka-X Are you saying that these two modules have duplicate code ?
![image](https://github.com/apache/seatunnel/assets/29418975/1afd0a06-1ccf-41a6-b12e-58858ce8e959)

Reply:
yes

Reply:
@Hisoka-X The modification is completed, please check it ， thk